### ISSUE TITLE
Exported assets/images with 0 bytes

### ISSUE THREAD

**Issue by pilarmera** [WordPress/create-block-theme#280]
Hi!

When I export the theme with the plugin, the image files used in the templates are exported (as mentioned in #213) but they contain no info, 0 bytes.

To reproduce (WP 6.1.1, Create Block Theme 1.7.0)
- On this plugin, create a blank theme and activate
- On the footer template, add and Image block and upload to the library a png file

![imagen](https://user-images.githubusercontent.com/1857745/225651719-389e4abe-261d-46e6-bd45-89bf06fbc1d3.png)
 
- On this plugin, Export the theme
![imagen](https://user-images.githubusercontent.com/1857745/225651120-40eb70b6-bcae-45ec-95f4-9d39d18df291.png)

- The generated .zip file contains the image file in /assets/images, but its size is 0 bytes. So it fails to load when I try to upload this theme into another installation

![imagen](https://user-images.githubusercontent.com/1857745/225651336-02166b32-1066-430c-abe4-b1c7799206eb.png)

p.d. I'm quite new to FSE and so on, git included, so maybe I'm missing something, sorry in advance and thank you for this plugin, I'm learning a lot!

**Comment by matiasbenedetto** [comment]
Hi @decrecementofeliz :wave: 
Thanks for reporting this issue! I just tried adding an image to a footer template part and it worked as expected. So I'm thinking about what can be happening in your case.

Is the file hosted in the local WordPress media files or is it a linked file?
Have you tried with other image files to see if that the same problem occurs?
Which PHP version and server you are running?

It would be helpful also if you can record a screencast video adding the image in the editor and after that export the theme. Maybe it would be useful to spot details I cannot see in the screenshots.

**Comment by pilarmera** [comment]
That's fast :) Thank you
I have recorded the issue, trying with png, webp and jpg files

The server is quite standard for me, isn't it?:
Linux 5.10.0-0.deb9.20-amd64 x86_64
Apache
7.4.33 (Soporta os valores de 64 bits)

**Comment by matiasbenedetto** [comment]
Hi @decrecementofeliz :wave: thanks for the screencast.
It's possible that your PHP config is making the plugin fail to download the images.
You need to have `allow_url_fopen = On` in your PHP config.
Could you check if you have that setting enabled in your `php.ini` file, please?
Let me know if you need help with that.

**Comment by pilarmera** [comment]
Hi again, allow_url_fopen is ON
![imagen](https://user-images.githubusercontent.com/1857745/226105985-88acbc00-fd3e-4cf1-a0bb-0678284f4210.png)

I'm wondering now, if I should I have the Gutenberg plugin installed and activated, but I have tested right now and the problem persists, no clue :(

**Comment by matiasbenedetto** [comment]
@decrecementofeliz I have another question:

Have you tried adding google fonts to your theme using the plugin? Did that work as expected?
If you haven't tried that yet: could you try it and let me know if it works, please?

I'm trying to figure out what can be causing the 0-byte image files.

**Comment by pilarmera** [comment]
Hi Matias, thanks for keeping up.

Fonts are working ok, I had already uploaded my custom local font and now I have added Google fonts, too, both type of files get correctly downloaded. But images still don't

![image](https://user-images.githubusercontent.com/1857745/227026969-21fc752c-4bf5-4673-819b-fb8bf44063ef.png)

**Comment by matiasbenedetto** [comment]
> both type of files get correctly downloaded

Good to know!

@decrecementofeliz: Does the image problem happen only when you are creating a .zip file (with the **export** or **clone** options of the plugin) or when you overwrite the local theme (with the **overwrite** option of the plugin) too?

**Comment by pilarmera** [comment]
It happens either way. 

On a different WP installation, on a different server from previous site I ran this: 

- Activated TT3 theme, edited the Footer part and inserted there an image block. 
- Then on the plugin I tried all possible options, in this order: Export (twentytwentythree.zip), Create child of TT3 (tt3child.zip), Clone TT3 (tt3clone.zip), Overwrite (and then Export again, twentytwentythree-2.zip). 

All these files contained 0-bytes images on assets/images.

**Comment by matiasbenedetto** [comment]
Hi @decrecementofeliz, I submitted a potential fix for this issue: https://github.com/WordPress/create-block-theme/pull/291
It would be helpful if you could try the fix. To do that, you can uninstall your current version of the plugin and download the plugin version linked in #291 .

**Comment by pilarmera** [comment]
Hi Matias, I've tested the fix and now it works!
![image](https://user-images.githubusercontent.com/1857745/228629654-5b0fc8b7-92c8-47ec-90a0-aba9cff6a2b4.png)

To make sure about everything I tried it all over in a new fresh installation: 

- First with the bug-version of the plugin I created TT3 child theme, added an image on the footer, exported the theme > 0 bytes on asset file. 
- Then I deleted the plugin, uploaded the fix-version of the plugin and exported the theme again > image file is saved correctly

p.d. I had to upload and uncompress the .zip file through FTP because through the WP admin it warned me about potential malware on the file.

Let me know if you want to me to test anything else about #291 and thanks!
